grpc/src/main/java/com/linecorp/armeria/server/grpc/AbstractUnframedGrpcService.java

```diff
@@ -182,6 +186,12 @@ static void deframeAndRespond(ServiceRequestContext ctx,
         final HttpHeaders trailers = !grpcResponse.trailers().isEmpty() ?
                                      grpcResponse.trailers() : grpcResponse.headers();
         final String grpcStatusCode = trailers.get(GrpcHeaderNames.GRPC_STATUS);
+        if (grpcStatusCode == null) {
+            PooledObjects.close(grpcResponse.content());
+            res.completeExceptionally(new NullPointerException("grpcStatusCode must not be null"));
+            logger.warn("A gRPC response must have the {} header.", GrpcHeaderNames.GRPC_STATUS);
```

Comment:
Should we add more context to the log message? It seems difficult to know which response omits `GrpcHeaderNames.GRPC_STATUS`.
```suggestion
            logger.warn("{} A gRPC response must have the {} header. response: {}",
                    ctx, GrpcHeaderNames.GRPC_STATUS, grpcResponse);
```